### ISSUE TITLE
Cannot access attribute "__orig_class__"

### ISSUE THREAD

**Issue by Andrej730** [microsoft/pyright#7896]
**Describe the bug**
It seems that pyright doesn't recognize `__orig_class__` dunder attribute that's available on instances from classes created with `Generic`. This attribute is probably will be used very rarily (and [for something hacky](https://stackoverflow.com/questions/60584388/how-to-check-typevars-type-at-runtime)), just wanted to report to make sure it's documented.

**Code or Screenshots**
```python
from __future__ import annotations
from typing import Generic, TypeVar

# fmt: off
T = TypeVar("T")

class BaseContainer(Generic[T]):
    def show_generic_type(self):
        # Cannot access attribute "__orig_class__" for class "BaseContainer[T@BaseContainer]*"
        # Attribute "__orig_class__" is unknown  (reportAttributeAccessIssue)
        print(self.__orig_class__)

BaseContainer[int]().show_generic_type()
```

**VS Code extension or command-line**
Tested on https://pyright-play.net/

**Comment by erictraut** [comment]
Yes, this is intended. `__orig_class__` is considered an internal implementation detail used to implement support for generics at runtime. You generally should not access it unless you're writing specialized code like a runtime type checker.

Note that `__orig_class__` doesn't appear in the typeshed type definitions for the `object` class. If you can convince the maintainers of typeshed to add it, then it will work with pyright, but I don't think this is something that pyright (or other type checkers) should special-case.

I'll note that mypy also generates an error in the code above, as I would expect.

Closing as "won't fix" because I think pyright is working correctly here.
